### PR TITLE
Feat: Authenticate resources with jwt

### LINKED ISSUE #28: Setup authentication system

### PR DESCRIPTION
**Pull request by tungnt-580**
closes #28

### CODE REVIEW

File: config/passport.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: 'secret', // We should put it to .env
+
+}
+
+passport.use(new JwtStrategy(options, (jwtPayload, done) => {
+  User.findById(jwtPayload.id)
+    .exec()
+    .then((user) => {
+      if (user) next(null, user)
```
Comment (kevcomedia) [inline config/passport.js]: Did you mean `done`, not `next`?

Please run the linter before pushing. It detects undeclared identifiers in the code (like `next` here).

File: config/passport.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      else next(null, false)
+    })
+    .catch((err) => next(err))
```
Comment (kevcomedia) [inline config/passport.js]: `catch(done)` looks neater than `catch((err) => done(err))`, but pick whichever one you like.

File: config/passport.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const passport = require('passport')
+const passportJwt = require('passport-jwt')
+const User = require('../models/user')
+
+const { ExtractJwt, Strategy: JwtStrategy } = passportJwt
+const options = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: 'secret', // We should put it to .env
+
```
Comment (kevcomedia) [inline config/passport.js]: Please remove this blank line

**Review by kevcomedia — CHANGES_REQUESTED** [review]
You named the JwtStrategy callback as `done`, but you used `next` in the callback body.

File: config/passport.js (hunk 1 above)
Comment (tungnt-580) [reply config/passport.js]: yeah I forgot to run ESLint!

**Review by kevcomedia — APPROVED** [review]

**Pull request merged**